### PR TITLE
chore(mdx): use mdx instead within docs/api

### PR DESCRIPTION
**Pull request by thetaPC**
Issue URL: N/A

## What is the current behavior?

Some files are using the `.md` extension. Docusaurus v3 has [stated](https://docusaurus.io/docs/migration/v3#using-the-mdx-extension) that at some point that these files will no longer be parsed as expected and will parsed as CommonMark. They advice to use `.mdx` instead.

## What is the new behavior?

- `docs/api/` files have been switched from `.md` to `.mdx`
- Switched to the proper syntax for MDX comments.
- Updated paths to no longer reference `.md`

same for v5 and v6

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

I'm splitting these into smaller PRs to make it easier to review.

[API docs](https://ionic-docs-git-mdx-pt-2-ionic1.vercel.app/docs/api/action-sheet)

### CODE REVIEW

File: docs/api/buttons.mdx [hunk 1]
```diff
@@ -62,7 +62,7 @@ This feature is only available for iOS.
 
 :::
 
-<!-- Reuse the playground from the Title directory -->
+{/* Reuse the playground from the Title directory */}
```
Comment (thetaPC) [inline docs/api/buttons.mdx:65]: Switched to the proper syntax for MDX.

File: docs/api/grid.mdx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import EncapsulationPill from '@components/page/api/EncapsulationPill';
 <EncapsulationPill type="shadow" />
 
 
-The grid is a powerful mobile-first flexbox system for building custom layouts. It is composed of three units — a grid, [row(s)](row.md) and [column(s)](col.md). Columns will expand to fill the row, and will resize to fit additional columns. It is based on a 12 column layout with different breakpoints based on the screen size. The number of columns can be customized using CSS.
+The grid is a powerful mobile-first flexbox system for building custom layouts. It is composed of three units — a grid, [row(s)](row) and [column(s)](col). Columns will expand to fill the row, and will resize to fit additional columns. It is based on a 12 column layout with different breakpoints based on the screen size. The number of columns can be customized using CSS.
```
Comment (thetaPC) [inline docs/api/grid.mdx:21]: Updated paths to no longer reference `.md`

**Review by mapsandapps — CHANGES_REQUESTED** [review]
All the changes here look good, but I did have some questions. If these are in another PR or something, lmk:

probably relevant to this PR:
* docusaurus.config.js: change lines 42-43 from .md to .mdx? (i didn't look into what this is used for; maybe we need to match both .md and .mdx and return .mdx?)
* `_templates/README.md`: line 24

also, the links in these to /api/foo.md seem to work, but maybe they should be changed for consistency?
* cross-platform.md
* structure.md
* css-shadow-parts.md
* css.variables.md
* 4-0.md

probably not relevant to this PR, but is there a plan to change these things to mdx?
* package.json: use mdx in `npm run spellcheck` command
* .editorconfig: use mdx instead of md (might need no/different rules for mdx, i didn't look into it)
* CONTRIBUTING.md: change filenames
* other lines in the docusaurus.config.js

(i marked request changes, but there may not actually be any changes needed, just let me know what's up 😄)

**Comment by thetaPC** [comment]
@mapsandapps in regards to the last part in your comment: I plan to do that in a separate PR.

**Review by mapsandapps — APPROVED** [review]
Thanks!

**Pull request merged**